### PR TITLE
Fix animation issues with dismiss read

### PR DESCRIPTION
**Pull request by CTalvio**
## Pull Request Description

Fixes dismiss animation often failing to play when used after many new posts have been marked read. Also dismiss is now staggered. Cool!

## Screenshots / Recordings

[Kazam_screencast_00002.webm](https://github.com/thunder-app/thunder/assets/4365015/a9586998-0e5d-4268-8024-3a540e8172d4)

## Checklist
N/A

### CODE REVIEW

**Comment by CTalvio** [comment]
Before:

[Kazam_screencast_00001.webm](https://github.com/thunder-app/thunder/assets/4365015/7f8db95d-3078-4d10-987b-363948c629c7)

**Review by micahmo — APPROVED** [review]
Nice, I had noticed this too. LGTM

File: lib/community/widgets/post_card_list.dart [hunk 1]
```diff
@@ -359,15 +360,25 @@ class _PostCardListState extends State<PostCardList> with TickerProviderStateMix
 
   Future<void> dismissRead() async {
     if (widget.postViews != null) {
-      setState(() {
-        showRead = false;
-      });
-      await Future.delayed(const Duration(milliseconds: 400));
+      if (!(widget.postViews!.length > 15)) {
```
Comment (hjiangsu) [inline lib/community/widgets/post_card_list.dart:363]: Just a quick question - why are there two calls to `widget.onScrollEndReached();`? One here and one on line 380

File: lib/community/widgets/post_card_list.dart [hunk 2]
```diff
@@ -359,15 +360,25 @@ class _PostCardListState extends State<PostCardList> with TickerProviderStateMix
...
+      }
+      for (var post in widget.postViews!) {
+        if (post.postView.read) {
+          setState(() {
+            toRemoveSet.add(post.postView.post.id);
+          });
+          await Future.delayed(const Duration(milliseconds: 60));
+        }
+      }
+      await Future.delayed(const Duration(milliseconds: 800));
```
Comment (hjiangsu) [inline lib/community/widgets/post_card_list.dart:374]: What is this for again? Is it to wait for all the animations to finish before resetting the state?

**Review by hjiangsu — COMMENTED** [review]
Just left a few comments!

File: lib/community/widgets/post_card_list.dart (hunk 2 above)
Comment (CTalvio) [reply lib/community/widgets/post_card_list.dart:374]: Yes. For some reason it does not wait for the loop to finish. There's probably a better way to do this, but I didn't figure one out yet.

Comment (CTalvio) [reply lib/community/widgets/post_card_list.dart:381]: And the 60 ms wait is what causes the dismissals to be staggered.

File: lib/community/widgets/post_card_list.dart (hunk 1 above)
Comment (hjiangsu) [reply lib/community/widgets/post_card_list.dart]: Ahh okay - I think we can make the second call a bit more robust in this case. I actually had to do this to resolve a issue where posts would not load on tall screens.

In my case, I had to do it when the state was first initiated, but for your case, we can make do without `WidgetsBinding.instance.addPostFrameCallback((_) {}`

```dart
// Check to see if the initial load did not load enough items to allow for scrolling to occur and fetches more items
WidgetsBinding.instance.addPostFrameCallback((_) {
  bool isScrollable = _scrollController.position.maxScrollExtent > _scrollController.position.viewportDimension;

  if (context.read<CommunityBloc>().state.hasReachedEnd == false && isScrollable == false) {
    widget.onScrollEndReached();
  }
});
```

File: lib/community/widgets/post_card_list.dart (hunk 2 above)
Comment (hjiangsu) [reply lib/community/widgets/post_card_list.dart:381]: Ahh okay okay, sounds good. I do think theres a better way but we can leave it as it is for now!

File: lib/community/widgets/post_card_list.dart (hunk 1 above)
Comment (CTalvio) [reply lib/community/widgets/post_card_list.dart]: Done

**Pull request merged**